### PR TITLE
Add num-most-recent-versions-to-keep property

### PR DESCRIPTION
**Pull request by hhanh**
**num-most-recent-versions-to-keep** is the minimum number of latest versions to keep when deleting old versions.
The property is used in combination with **num-old-versions-to-delete**. When deleting old versions using **num-old-versions-to-delete** property, making sure that at least **num-most-recent-versions-to-keep** of most recent versions to be kept.

### CODE REVIEW

**Comment by rvdh** [comment]
@jclem @cdb @trent-j Can someone please look at this? This would make the action a lot more useful! Thanks.

**Comment by hazcod** [comment]
REALLY would like this feature, it's crazy to remove x packages without a minimum to keep.

**Review by docwhat — APPROVED** [review]

**Comment by lajos-lorand** [comment]
Hi, Any news when will be this feature available? It would very useful.

**Comment by phgroe** [comment]
Upvote! There is one failing "Check licenses" ... check, otherwise nothing seems to be in the way. Changes in this PR were approved by @docwhat. Who can repair the failing check and merge this, pretty please?

**Comment by hhanh** [comment]
Hi,

I fixed the license checking issue. Anyone with write access could merge ?

Bests,

**Comment by dthpth** [comment]
@andymckay @thboop @trent-j could you please take a look? 
It could close two requests (#5 and #17) with 90 and 40 upvotes accordingly. thanks!

**Comment by hazcod** [comment]
@hhanh perhaps nice to include in the readme:
```yaml
name: Package Cleanup

on:
  push:
    branches: [master]

jobs:

  cleanpackages:
  -
    name: frontend package
    uses: actions/delete-package-versions@v1
    with:
      package-name: "foo/bar"
      num-old-versions-to-delete: 5
      num-most-recent-versions-to-keep: 5
```

**Comment by e-moshaya** [comment]
any update on this? I'd like to start using this change please

File: src/delete.ts [hunk 1]
```diff
@@ -13,9 +13,17 @@ export function getVersionIds(input: Input): Observable<string[]> {
...
+      input.numOldVersionsToDelete + input.numMostRecentVersionsToKeep,
       input.token
-    ).pipe(map(versionInfo => versionInfo.map(info => info.id)))
+    ).pipe(
+      map(versionInfo => {
+        const numberVersionsToDelete =
+          versionInfo.length - input.numMostRecentVersionsToKeep
+        return numberVersionsToDelete <= 0
+          ? []
+          : versionInfo.slice(0, numberVersionsToDelete).map(info => info.id)
```
Comment (ashenm) [inline src/delete.ts:24]: We could omit the supplementary ternary here and simply make this as follows; in the event of `[0, -n]` the `slice` should evaluate to an empty array.
 
```suggestion
        return versionInfo.slice(0, numberVersionsToDelete).map(info => info.id)
```

**Review by ashenm — CHANGES_REQUESTED** [review]

File: src/delete.ts (hunk 1 above)
Comment (hhanh) [reply src/delete.ts:24]: As I understand,  end index being negative number indicates an offset from the end of the sequence. So, slice with negative end index will not result an empty array.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/slice

```
let a = [1,2,3,4,5];
console.log(a.slice(0,-2)); // print [1,2,3]
```

Comment (ashenm) [reply src/delete.ts:24]: Shoot! Missed `Math.max` call! I have updated the comment :yum:

Comment (ashenm) [reply src/delete.ts:24]: Also lets get remove that `numberVersionsToDelete` as well since we are referencing it only once with the above change

**Comment by eberkund** [comment]
Hey, any updates on this?

**Comment by c4710n** [comment]
> Hey, any updates on this?

Try `smartsquaregmbh/delete-old-packages`.

**Comment by hazcod** [comment]
Ping..

**Comment by stephan-strate** [comment]
Can we get this integrated? Any status update?

**Comment by NamrataJha** [comment]
This feature has been released with the latest version `actions/delete-package-versions@v2`. Closing this PR.

**Pull request closed without merging**